lib/vanagon/project.rb

```diff
@@ -286,12 +297,51 @@ def get_preinstall_actions(pkg_state)
       end
     end
 
+    # Collects the install trigger scripts for the project for the specified packing state
+    #
+    # @param pkg_state [String] the package state we want to run the given scripts for.
+    #   Can be one of the 'install' or 'upgrade'
+    # @return [Hash] of scriptlets to execute during the pkg_state (install or upgrade)
+    #   there can be more than one script for each package (key)
+    def get_install_trigger_scripts(pkg_state)
+      triggers = Hash.new { |hsh, key| hsh[key] = [] }
+      check_pkg_state_string(pkg_state)
+      pkgs = @components.map(&:install_triggers).flatten.compact.select { |s| s.pkg_state.include? pkg_state }
```

Comment:
Suggestion: Use `#flat_map` and the `#components` accessor instead of `#map.flatten` and `@components`:

```ruby
pkgs = components.flat_map(&:install_triggers).compact.select { |s| s.pkg_state.include? pkg_state }
```